### PR TITLE
New Azure Cognitive Search API version 2023-07-01-Preview

### PR DESCRIPTION
**Pull request by Careyjmac**
# Data Plane API - Pull Request
Introducing new preview Azure Cognitive Search data plane API version 2023-07-01-Preview. This version includes the following changes from the previous 2021-04-30-Preview API version:

- Deprecates the TextAnalyticsV2 EntityRecognition and Sentiment skills. These skills will become non-functional in August 2024, but we are preemptively blocking any upserts with skillsets containing these skills starting in this API version.
- Various changes to support the upcoming vector search feature.
- Various changes to improve the existing semantic search feature.
- Adding user identity configuration to the WebApiSkill, knowledge store and indexer cache features. These features actually already existed in the 2021-04-30-Preview API version but [the PR to add them to the swagger was never merged](https://github.com/Azure/azure-rest-api-specs/pull/16592) for some reason.

See the [private PR](https://github.com/Azure/azure-rest-api-specs-pr/pull/11998) for a better diff.

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [X] a private preview
- [X] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json:299]: should speller and all be removed from here?

Comment (BevLoh) [reply specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json:299]: same in the post version as well

Comment (Careyjmac) [reply specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json:299]: Good catch, I removed the value descriptions in the other PR but I guess I forgot to remove them here.

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (tg-msft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6055]: Nit - if this is just the name of one of the configs in the array below, it’d be great to suggest that in the docs:
```suggestion
          "description": "Allows you to set the name of a default semantic configuration in your index, making it optional to pass it on as a query parameter every time."
```

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (tg-msft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6155]: We should make this an extensible enum in your next release, but for now can we do something like:
```suggestion
          "description": "The name of the kind of algorithm being configured for use with vector search  Only `hnsw` is supported in the current preview.",
```

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (tg-msft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:8665]: Nit - I don't think this will render in the docs.  It might be nice to also update the description to say it's deprecated and point users at a replacement.

**Review by tg-msft — APPROVED** [review]

File: specification/search/data-plane/Azure.Search/readme.md
Comment (tg-msft) [inline specification/search/data-plane/Azure.Search/readme.md:226]: Nit - you need to close this code block.
``````suggestion
``` yaml $(tag) == 'package-2023-07-searchindex-preview'
input-file:
- preview/2023-07-01-Preview/searchindex.json
```

### Tag: package-2021-04-30-preview
``````

File: specification/search/data-plane/Azure.Search/readme.md
Comment (tg-msft) [inline specification/search/data-plane/Azure.Search/readme.md:187]: Let's remove this for now.  We've got the tracking bug filed at https://github.com/Azure/azure-rest-api-specs/issues/23807 which I think is enough to move forward.
```suggestion
```

**Review by tg-msft — APPROVED** [review]

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json
Comment (weidongxu-microsoft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchindex.json:6]: I think `preview` is usually lower-case?
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning-date-based-versioning

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (weidongxu-microsoft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6154]: Should it be an enum with model-as-string=true?

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (weidongxu-microsoft) [inline specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6161]: Should `kind` be a discriminator, and `hnswParameters` be in the model inherit `VectorSearchAlgorithmConfiguration` (so that other type/kind with different parameters can be added in future)?

This need to be solved before GA. Preview is fine.

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (Careyjmac) [reply specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6154]: Ted also made this comment I think. I'm going to leave this as is for now so we can get this in and unblock the //build efforts, but we can definitely consider changing this for GA.

File: specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json
Comment (Careyjmac) [reply specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6161]: I think there have been extensive discussions around this already, and we are opting not to go this route, but I could be wrong. Regardless, as you mentioned we can move forward as is for preview.

Comment (tg-msft) [reply specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json:6161]: Yes, Weidong is correct here.  I think this was accidentally flattened into the base class, but we can fix that before the next release.

**Review by tg-msft — APPROVED** [review]
Everything looks great to me.  The last fixes to make the spec more accurate for what the service does look correct and are not breaking changes.  The only suppressions added were removed in later commits so that's fine too.  The policheck errors need to be surpressed, but are blocked on https://github.com/Azure/azure-sdk-tools/issues/6091.  This is ready to merge.

**Pull request merged**
